source/kernel/src/registry/mod.rs

```diff
@@ -259,16 +306,21 @@ impl Registry {
         if self.items.iter().any(|i| i.key == RD::UUID) {
             return Err(RegistrationError::UuidAlreadyRegistered);
         }
-        self.items
+        let result = self
+            .items
             .push(RegistryItem {
                 key: RD::UUID,
                 value: RegistryValue {
                     req_resp_tuple_id: RD::type_id().type_of(),
                     req_prod: kch.clone().type_erase(),
                     req_deser: Some(map_deser::<RD>),
+                    service_id: self.counter,
                 },
             })
-            .map_err(|_| RegistrationError::RegistryFull)
+            .map_err(|_| RegistrationError::RegistryFull)?;
+        info!(uuid = ?RD::UUID, service_id = self.counter, "Registered");
+        self.counter = self.counter.wrapping_add(1);
```

Comment:
wonder if we should maybe log something in the event that this wraps, just so that a reader of the logs will know that they may see the same IDs repeat

Reply:
Kicking this can down the road.